cmd/playground/interactive_window.go

```diff
@@ -0,0 +1,213 @@
+			case "TruncateTo":
+				m.state.windowv2.TruncateTo = m.state.DecrementTruncate()
+			case "year", "month", "day", "hour", "minute":
+				m.state.DecrementTime(m.cursor)
+			}
+		// if not the arrow keys then the input must be given to the size or the off set hence we update them accordingly
+		default:
+			// to get updated the input fields must be focussed since we handle the Focus and Blur we can directly update them here
+			m.Sizeinput, _ = m.Sizeinput.Update(msg)
+			m.Offsetinput, _ = m.Offsetinput.Update(msg)
+		}
+	}
+	// update the values of the window versions in state for every modification from input
+	m.updateWindowVersions()
+	return m, nil
+}
+
+// we genarate a cursor if the value of the cursor matches with the current field name, else we pass a blank instead
+func (m model) genarateFields(position string, value string) string {
+	if m.cursor == position {
+		var s strings.Builder
+		s.WriteString("[")
+		s.WriteString(value)
+		s.WriteString("]")
+		return s.String()
+	}
+	return " " + value
+}
+
+// we genare a string representing the sechduled time , which also adds a cursor if it is pointing to the any of the fields in sechdueld date
+func (m model) genarateSechduledDateView() string {
+	var s strings.Builder
+	s.WriteString(m.genarateFields("hour", strconv.Itoa(m.state.sechduledTime.Hour())))
+	s.WriteString(":")
+	s.WriteString(m.genarateFields("minute", strconv.Itoa(m.state.sechduledTime.Minute())))
+	s.WriteString(m.genarateFields("day", strconv.Itoa(m.state.sechduledTime.Day())))
+	s.WriteString(m.genarateFields("month", m.state.sechduledTime.Month().String()))
+	s.WriteString(m.genarateFields("year", strconv.Itoa(m.state.sechduledTime.Year())))
+	return s.String()
+}
+
+// this will update the values of Size and offset of the window versions present in state to the new values taken from the input
+func (m *model) updateWindowVersions() {
+	// .Value() here represents the string that is present inside the text field
+	m.state.windowv1.Size = m.Sizeinput.Value()
+	m.state.windowv2.Size = m.Sizeinput.Value()
+	m.state.windowv1.Offset = m.Offsetinput.Value()
+	m.state.windowv2.Offset = m.Offsetinput.Value()
+	m.state.windowv1.TruncateTo = m.state.windowv2.TruncateTo
+}
+
+// this will  be invoked for every update
+func (m model) View() string {
+	var s strings.Builder
+	s.WriteString("Size            ")
+	s.WriteString(m.Sizeinput.View())
+	s.WriteString("\n")
+	s.WriteString("Offset          ")
+	s.WriteString(m.Offsetinput.View())
+	s.WriteString("\n")
+	s.WriteString("TruncateTo       ")
+	s.WriteString(m.genarateFields("TruncateTo", m.state.windowv2.TruncateTo))
+	s.WriteString("\n")
+	s.WriteString("sechduled date   ")
+	s.WriteString(m.genarateSechduledDateView())
+	s.WriteString("\n\n\n             ") // extra space added for design reasons
+	// calculate the value of dstart and dend from the imported window versions
+	dstartv1, dendv1, err := m.state.windowv1.GetTimeRange(m.state.sechduledTime)
```

Comment:
use `dStartV1` instead of `dstartv1` (this also for the `end` and the version 2 as well)